tests/RestfulVariableTestCase.test

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * @file
+ * Contains RestfulDbQueryTestCase.
+ */
+
+class RestfulVariableTestCase extends DrupalWebTestCase {
+
+  /**
+   * The name of the test table.
+   *
+   * @var string
+   */
+  protected $tableName = 'restful_test_variable';
+
+  /**
+   * Provides information about the test class.
+   */
+  public static function getInfo() {
+    return array(
+      'name' => 'Variable',
+      'description' => 'Test the variable data provider.',
+      'group' => 'Restful',
+    );
+  }
+
+  /**
+   * Operations before the testing begins.
+   */
+  function setUp() {
+    parent::setUp('restful_example');
+  }
+
+  /**
+   * Test authenticating a user.
+   */
+  function testCrudOperations() {
+    // Populate the table with some variables.
+    for ($i = 0; $i < 6; $i++) {
+      variable_set('variable_' . $i, array('test_data' => intval(mt_rand(1, 100))));
+    }
+
+    $this->assertTrue(variable_get('variable_5'), 'The variables have been set.');
+
+    // Get the handler.
+    $handler = restful_get_restful_handler('variables');
+
+    // Testing read
+    $results = $handler->get('variable_5');
+    $result = $results[0];
+    $this->assertEqual($result['variable_name'], 'variable_5', 'The variable name was successfully retrieved.');
+    $this->assertNotNull($result['variable_value'], 'The variable value was successfully retrieved.');
+
+    // Testing list.
+    $results = $handler->get();
+    $inResults = FALSE;
+    foreach ($results as $result) {
+      if ($result['variable_name'] == 'variable_5') {
+        $inResults = TRUE;
+      }
+    }
+    $this->assertTrue($inResults, 'All the content listed successfully.');
+
+    // Testing sort.
+    variable_set('zzz', array('some_property' => $this->randomName()));
```

Comment:
Imagine a site that has a variable named `zzz_counter` (for the imaginary zzz module). This can lead to false positives. We should get all the variable names, then get the last (in strcmp order) and generate one that would compare after that.